File: 5.1-martian/martian0/src/states/game_states/PlayState.py
"""
ISPPJ1 2023
Study Case: Super Martian (Platformer)

Author: Alejandro Mujica
alejandro.j.mujic4@gmail.com

This file contains the class PlayState.
"""
from typing import Dict, Any

import pygame

from gale.state import BaseState

import settings


class PlayState(BaseState):
    def enter(self, **enter_params: Dict[str, Any]) -> None:
        self.level = enter_params.get("level", 1)
        self.tilemap = None
        settings.LevelLoader().load(self, settings.TILEMAPS[self.level])

    def render(self, surface: pygame.Surface) -> None:
        self.tilemap.render(surface)

File: 7.1-ultimate_fantasy/uf1/src/states/game/PlayState.lua
--[[
    ISPPJ1 2023
    Study Case: Ultimate Fantasy (RPG)

    Author: Alejandro Mujica
    alejandro.j.mujic4@gmail.com

    This class contains the class PlayState.
]]
PlayState = Class{__includes = BaseState}

function PlayState:init(def)
    world = World(def)
end

function PlayState:update(dt)
    world:update(dt)
end

function PlayState:render()
    world:render()
end
